bin/oref0-upload-profile.js

```diff
@@ -144,14 +144,22 @@ if (!module.parent) {
 
             var time = target_entry.start.substring(0, 5);
             var seconds = parseInt(time.substring(0, 2)) * 60 * 60 + parseInt(time.substring(3, 5)) * 60;
-            var low_value = Math.round(target_entry.low);
-            var high_value = Math.round(target_entry.high);
-
-            if (new_profile.units === 'mmol' && profiledata.bg_targets.units === 'mg/dL') {
-                low_value = +(Math.round(target_entry.low / 18 + 'e+1') + 'e-1');
-                high_value = +(Math.round(target_entry.high / 18 + 'e+1') + 'e-1');
+            var low_value = target_entry.low;
+            var high_value = target_entry.high;
+            var conversionFactor = 1;
+            var decimals = new_profile.units === 'mmol' ? 10 : 1;
+
+            if (new_profile.units !== profiledata.bg_targets.units) {
+                // if the new profile is using a different unit than our BG targets, then set the conversion factor accordingly
+                conversionFactor = (new_profile.units == 'mmol' ? 0.055 : 18);
```

Comment:
It’s not immediately obvious what you’re doing here. Is this to support both cases where the units are mismatched, and multiply or divide appropriately?

Either way, it’d probably best to add an explanatory comment here.

Reply:
Yeah, this is to handle cases where the autotune profile is in different units from the Nightscout profile. The old code essentially did the same thing but only handled converting autotune values from mg/dL to mmol/L if the Nightscout profile was using mmol/L: `if (new_profile.units === 'mmol' && profiledata.bg_targets.units === 'mg/dL')` 

I'll add comments to explain this.